src/content/toys.ts

```diff
@@ -32,17 +32,98 @@ function makeBlock() {
     overlaydiv.style.position = "fixed"
     overlaydiv.style.display = "block"
     overlaydiv.style.width = String(window.innerWidth)
-    overlaydiv.style.height = String(window.innerHeight)
-    overlaydiv.style.top = "0"
-    overlaydiv.style.left = "0"
-    overlaydiv.style.right = "0"
-    overlaydiv.style.bottom = "0"
+    overlaydiv.style.height = String(document.documentElement.scrollHeight)
+    overlaydiv.style.top = "0px"
+    overlaydiv.style.bottom = "0px"
+    overlaydiv.style.left = "0px"
+    overlaydiv.style.right = "0px"
     overlaydiv.style.zIndex = "1000"
     overlaydiv.style.opacity = "0.5"
     document.body.appendChild(overlaydiv)
     return overlaydiv
 }
 
+export function drawable() {
+    eraser = false
+    make_drawable(makeBlock())
+}
+
+const clickX = []
+const clickY = []
+const clickDrag = []
+let ink
+
+let eraser = false;
+export function eraser_toggle() {
+    eraser = !eraser
+}
+
+function addClick(x, y, dragging) {
+    clickX.push(x)
+    clickY.push(y)
+    clickDrag.push(dragging)
+}
+
+function redraw(context) {
+    if(eraser) {
+        context.globalCompositeOperation = "destination-out"
+        context.lineWidth = 18
+    } else {
+        context.globalCompositeOperation = "source-over"
+        context.lineWidth = 3
+    }
+    context.strokeStyle = "#000000"
+    context.lineJoin = "miter"
+    for(let i=0; i < clickX.length; i++) {
+        context.beginPath()
+        if(clickDrag[i] && i){
+            context.moveTo(clickX[i-1], clickY[i-1])
+        } else {
+            context.moveTo(clickX[i]-1, clickY[i])
+        }
+        context.lineTo(clickX[i], clickY[i])
+        context.closePath()
+        context.stroke()
+    }
+}
+function handleDown(e, context){
+    ink = true
+    addClick(e.pageX, e.pageY, false)
+    redraw(context)
+    e.preventDefault()
+    e.stopPropagation()
+}
+function handleUp(e){
+    ink = false
+    clickX.length = 0
+    clickY.length = 0
+    clickDrag.length = 0
+    e.stopPropagation()
+    e.preventDefault()
+}
+function handleMove(e, context) {
+    if(ink){
+        addClick(e.pageX, e.pageY, true);
+        redraw(context);
+    }
+    e.preventDefault()
+    e.stopPropagation()
+}
+function make_drawable(overlaydiv) {
+    overlaydiv.style.position = "absolute"
+    overlaydiv.style.opacity = "0.8"
+    const c = document.createElement("canvas")
+    overlaydiv.appendChild(c)
+    const context = c.getContext("2d")
+    // making the canvas full screen
+    c.height = document.documentElement.scrollHeight
+    c.width = window.innerWidth*0.98  // workaround to fix canvas overflow
+    c.style.touchAction = "none" // for pen tablet to work
+    c.addEventListener("pointerdown", e=>handleDown(e,context))
+    c.addEventListener("pointerup", handleUp)
+    c.addEventListener("pointermove", e=>handleMove(e,context))
```

Comment:
When we do #3613 these event listeners will need to be removed

(no action needed now - this comment should appear in the issue and remind us)